packages/@glimmer/runtime/lib/compiled/expressions/concat.ts

```diff
@@ -24,7 +24,9 @@ export function createConcatRef(partsRefs: Reference[]) {
 }
 
 function castToString(value: Dict) {
-  if (typeof value.toString !== 'function') {
+  if (typeof value === 'string') {
```

Comment:
if value is already string, likely we don't need to `toString` it

Reply:
The type said this takes a `Dict`, is the type wrong?

Reply:
it's just ordering, we return most relevant case first

Reply:
as I see this code used inside `createConcatRef`, and it's  seems we likely have strings here..

Reply:
Yeah it seems like the type is wrong then, probably should have been `unknown` if we had no idea what's in it

Reply:
Can you change the type too?

Reply:
@chancancode sure, will do it!

Reply:
done

Reply:
The correct type would be `unknown`

Reply:
@chancancode in this case we need to add extra check for `object` 
<img width="731" alt="image" src="https://github.com/glimmerjs/glimmer-vm/assets/1360552/6daaf027-0af7-4676-8693-4fb91c108a71">

Reply:
ahh, I see, because we already checked for `null` and `undefined` outside the call. yeah, ok in that case `string | object` is probably good enough